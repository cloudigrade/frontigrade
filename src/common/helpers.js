const generateId = prefix => `${prefix || 'generatedid'}-${Math.ceil(1e5 * Math.random())}`;

const getLocale = (messages = {}) => {
  const locale =
    process.env.REACT_APP_LOCALE ||
    (window.navigator.languages && window.navigator.languages[0]) ||
    window.navigator.language ||
    window.navigator.userLanguage;

  return locale.match(/^[A-Za-z]+-[A-Za-z]+$/) && !messages[locale] ? locale.split('-')[0] : locale;
};

const noop = Function.prototype;

const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state }; // eslint-disable-line prefer-const

  if (!state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (reset && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  }

  return obj;
};

const DEV_MODE = process.env.REACT_APP_ENV === 'development';

const FULFILLED_ACTION = base => `${base}_FULFILLED`;

const PENDING_ACTION = base => `${base}_PENDING`;

const REJECTED_ACTION = base => `${base}_REJECTED`;

export const helpers = {
  generateId,
  getLocale,
  noop,
  setStateProp,
  DEV_MODE,
  FULFILLED_ACTION,
  PENDING_ACTION,
  REJECTED_ACTION
};

export default helpers;
