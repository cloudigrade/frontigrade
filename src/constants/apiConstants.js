const API_ACCOUNT_ARN = 'account_arn';
const API_ACCOUNT_NAME = 'name';
const API_RESOURCE_TYPE = 'resourcetype';
const API_AWS_ACCOUNT_ID = 'aws_account_id';

const apiTypes = {
  API_ACCOUNT_ARN,
  API_ACCOUNT_NAME,
  API_RESOURCE_TYPE,
  API_AWS_ACCOUNT_ID
};

export { apiTypes as default, apiTypes, API_ACCOUNT_ARN, API_ACCOUNT_NAME, API_RESOURCE_TYPE, API_AWS_ACCOUNT_ID };
