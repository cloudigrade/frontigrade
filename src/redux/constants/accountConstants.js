const ADD_ACCOUNT = 'ADD_ACCOUNT';
const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
const DELETE_ACCOUNTS = 'DELETES_ACCOUNTS';
const GET_ACCOUNT = 'GET_ACCOUNT';
const GET_ACCOUNTS = 'GET_ACCOUNTS';
const ADD_ACCOUNT_WIZARD = 'ADD_ACCOUNT_WIZARD';
const ADD_ACCOUNT_WIZARD_STEP_POLICY = 'ADD_ACCOUNT_WIZARD_STEP_POLICY';
const INVALID_ACCOUNT_WIZARD_STEP_POLICY = 'INVALID_ACCOUNT_WIZARD_STEP_POLICY';
const ADD_ACCOUNT_WIZARD_STEP_ROLE = 'ADD_ACCOUNT_WIZARD_STEP_ROLE';
const INVALID_ACCOUNT_WIZARD_STEP_ROLE = 'INVALID_ACCOUNT_WIZARD_STEP_ROLE';
const ADD_ACCOUNT_WIZARD_STEP_ARN = 'ADD_ACCOUNT_WIZARD_STEP_ARN';
const INVALID_ACCOUNT_WIZARD_STEP_ARN = 'INVALID_ACCOUNT_WIZARD_STEP_ARN';
const ADD_ACCOUNT_SHOW = 'ADD_ACCOUNT_SHOW';
const EDIT_ACCOUNT_SHOW = 'EDIT_ACCOUNT_SHOW';
const UPDATE_ACCOUNT_HIDE = 'UPDATE_ACCOUNT_HIDE';

export {
  ADD_ACCOUNT,
  DELETE_ACCOUNT,
  DELETE_ACCOUNTS,
  GET_ACCOUNT,
  GET_ACCOUNTS,
  ADD_ACCOUNT_WIZARD,
  ADD_ACCOUNT_WIZARD_STEP_POLICY,
  INVALID_ACCOUNT_WIZARD_STEP_POLICY,
  ADD_ACCOUNT_WIZARD_STEP_ROLE,
  INVALID_ACCOUNT_WIZARD_STEP_ROLE,
  ADD_ACCOUNT_WIZARD_STEP_ARN,
  INVALID_ACCOUNT_WIZARD_STEP_ARN,
  ADD_ACCOUNT_SHOW,
  EDIT_ACCOUNT_SHOW,
  UPDATE_ACCOUNT_HIDE
};
