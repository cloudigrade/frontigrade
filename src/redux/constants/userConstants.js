const USER_CREATE = 'USER_CREATE';
const USER_DELETE = 'USER_DELETE';
const USER_INFO = 'USER_INFO';
const USER_LOGIN = 'USER_LOGIN';
const USER_LOGOUT = 'USER_LOGOUT';

export { USER_CREATE, USER_DELETE, USER_INFO, USER_LOGIN, USER_LOGOUT };
