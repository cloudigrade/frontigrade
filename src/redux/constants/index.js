import * as toastNotificationTypes from './toastNotificationConstants';
import * as userTypes from './userConstants';

const reduxTypes = {
  toastNotifications: toastNotificationTypes,
  user: userTypes
};

export { reduxTypes as default, reduxTypes, toastNotificationTypes, userTypes };
