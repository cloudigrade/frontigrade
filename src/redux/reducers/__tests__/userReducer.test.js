import { userReducers } from '../';
import { userTypes as types } from '../../constants';
import helpers from '../../../common/helpers';

describe('UserReducers', () => {
  it('should return the initial state', () => {
    expect(userReducers.initialState).toBeDefined();
  });

  it('should handle all defined error types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: helpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = userReducers(undefined, dispatched);

      expect(resultState).toMatchSnapshot();
    });
  });

  it('should handle all defined pending types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: helpers.PENDING_ACTION(value)
      };

      const resultState = userReducers(undefined, dispatched);

      expect(resultState).toMatchSnapshot();
    });
  });
});
