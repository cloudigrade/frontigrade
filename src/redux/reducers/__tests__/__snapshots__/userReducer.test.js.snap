// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReducers should handle all defined error types 1`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": true,
    "errorMessage": "MESSAGE",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;

exports[`UserReducers should handle all defined error types 2`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": true,
    "errorMessage": "MESSAGE",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;

exports[`UserReducers should handle all defined error types 3`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": true,
    "errorMessage": "MESSAGE",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;

exports[`UserReducers should handle all defined error types 4`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": true,
    "errorMessage": "MESSAGE",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;

exports[`UserReducers should handle all defined error types 5`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": true,
    "errorMessage": "MESSAGE",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;

exports[`UserReducers should handle all defined pending types 1`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": true,
  },
}
`;

exports[`UserReducers should handle all defined pending types 2`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": true,
  },
}
`;

exports[`UserReducers should handle all defined pending types 3`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": false,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": true,
  },
}
`;

exports[`UserReducers should handle all defined pending types 4`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": true,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;

exports[`UserReducers should handle all defined pending types 5`] = `
Object {
  "session": Object {
    "authorized": false,
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "pending": true,
  },
  "user": Object {
    "current": Object {},
    "error": false,
    "errorMessage": "",
    "fulfilled": false,
    "other": Object {},
    "pending": false,
  },
}
`;
